In [1]:
import statsmodels.stats.api as sms
import pandas as pd

# Parameters for the A/B test
conversion_rate_current = 0.05  # Current conversion rate
conversion_rate_new = 0.07  # Expected conversion rate with the new checkout process
effect_size = 0.2  # Given effect size for initial sample size calculation
alpha = 0.05  # Significance level
power = 0.8  # Desired power level

# Calculate the required sample size for the given effect size
sample_size_initial = sms.zt_ind_solve_power(effect_size=effect_size, alpha=alpha, power=power, alternative='two-sided')

# Analyze the impact of different effect sizes on required sample size
effect_sizes = [0.1, 0.2, 0.3, 0.4]
sample_sizes = [sms.zt_ind_solve_power(effect_size=es, alpha=alpha, power=power, alternative='two-sided') for es in effect_sizes]

# Create a DataFrame to display the results
sample_size_df = pd.DataFrame({
    'Effect Size': effect_sizes,
    'Sample Size': sample_sizes
})

# Display the initial sample size and the effect size analysis
sample_size_initial, sample_size_df


(392.4430232577885,
    Effect Size  Sample Size
 0          0.1  1569.772102
 1          0.2   392.443023
 2          0.3   174.419122
 3          0.4    98.110757)

To test whether the new checkout process (with an expected conversion rate of 7%) outperforms the current one (with a 5% conversion rate), the required sample size for each group, with an effect size of 0.2, is approximately 392 participants.

The smaller the effect size, the larger the sample needed to detect a difference. This is because small changes (like improving conversion rate by a tiny margin) are harder to spot and require more data to confirm. On the other hand, larger effects are more noticeable, so fewer samples are needed to be confident. In A/B testing, balancing effect size and sample size ensures you're not wasting resources (too many participants) or missing important insights (too few participants), helping you confidently make decisions to boost your bakery's sweet sales!